fe/fe-core/src/main/java/org/apache/doris/datasource/property/PropertyConverter.java

```diff
@@ -0,0 +1,337 @@
+            throw new IllegalArgumentException("Need endpoint with region: " + endpoint);
+        }
+        return endpointSplit[1];
+    }
+
+    public static Map<String, String> metaConvert(Map<String, String> props) {
+        if (props.containsKey(GlueProperties.ENDPOINT)
+                || props.containsKey(AWSGlueConfig.AWS_GLUE_ENDPOINT)) {
+            return convertToGlueProperties(props, GlueProperties.getCredential(props));
+        } else if (props.containsKey(DLFProperties.ENDPOINT)
+                || props.containsKey(DataLakeConfig.CATALOG_ENDPOINT)) {
+            return convertToDLFProperties(props, DLFProperties.getCredential(props));
+        }
+        return props;
+    }
+
+    /**
+     * Convert properties defined at doris to FE S3 client properties and BE S3 client properties
+     */
+    public static Map<String, String> storageConvert(Map<String, String> props) {
+        if (props.containsKey(S3Properties.Environment.ENDPOINT)) {
+            // compatible with the s3,obs,oss,cos when they use aws client.
+            return convertToS3EnvProperties(props, S3Properties.getCompatibleCredential(props));
+        } else if (props.containsKey(S3Properties.ENDPOINT)) {
+            return convertToS3Properties(props, S3Properties.getCredential(props));
+        } else if (props.containsKey(ObsProperties.ENDPOINT)) {
+            return convertToOBSProperties(props, ObsProperties.getCredential(props));
+        } else if (props.containsKey(OssProperties.ENDPOINT)) {
+            return convertToOSSProperties(props, OssProperties.getCredential(props));
+        } else if (props.containsKey(CosProperties.ENDPOINT)) {
+            return convertToCOSProperties(props, CosProperties.getCredential(props));
+        }
+        return props;
+    }
+
+
+    private static Map<String, String> convertToOBSProperties(Map<String, String> props,
+                                                              CloudCredential credential) {
+        Map<String, String> obsProperties = Maps.newHashMap();
+        obsProperties.put(OBSConstants.ENDPOINT, props.get(ObsProperties.ENDPOINT));
+        obsProperties.put("fs.obs.impl.disable.cache", "true");
+        if (credential.isWhole()) {
+            obsProperties.put(OBSConstants.ACCESS_KEY, credential.getAccessKey());
+            obsProperties.put(OBSConstants.SECRET_KEY, credential.getSecretKey());
+        }
+        if (credential.isTemporary()) {
+            obsProperties.put("fs.obs.session.token", credential.getSessionToken());
+        }
+        for (Map.Entry<String, String> entry : props.entrySet()) {
+            if (entry.getKey().startsWith(ObsProperties.OBS_FS_PREFIX)) {
+                obsProperties.put(entry.getKey(), entry.getValue());
+            }
+        }
+        BeAWSProperties.setBeAWSPropertiesFromObs(obsProperties);
+        return obsProperties;
+    }
+
+    private static Map<String, String> convertToS3EnvProperties(Map<String, String> properties,
```

Comment:
```suggestion
    private static Map<String, String> convertToS3HadoopFSProperties(Map<String, String> properties,
```

Reply:
S3HadoopFSProperties like this: "fs.s3a.xxx"
So named the function  as S3EnvProperties to distinguish it from S3HadoopFSProperties